### DISCUSSION TITLE
Could NOT find XCB (SteamOS)

### DISCUSSION THREAD

**Discussion by MvEerd** [polybar/polybar#2910] [Build Support] [answered]
Following up on #2908
After solving that error with `pacman -S alsa-lib`
I then solved the following errors;
- curl dep error with `pacman -S curl`
- libpulse error with `pacman -S libpulse`
- xcb-randr error with `pacman -S libxcb`
- xcb_cursor error with `pacman -S xcb-util-cursor`

I now get the following error;
```
==> Starting build()...
-- Couldn't locate ccache, disabling ccache...
-- Using supported compiler GNU-12.2.0
-- XCB[XCB]: Found component XCB
-- Could NOT find XCB_ICCCM (missing: XCB_ICCCM_INCLUDE_DIR) 
-- Could NOT find XCB_EWMH (missing: XCB_EWMH_INCLUDE_DIR) 
-- Could NOT find XCB_UTIL (missing: XCB_UTIL_INCLUDE_DIR) 
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/libxcb-image.so  
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find XCB (missing: XCB_ICCCM_FOUND XCB_EWMH_FOUND XCB_UTIL_FOUND)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  lib/xpp/cmake/FindXCB.cmake:249 (find_package_handle_standard_args)
  lib/xpp/CMakeLists.txt:28 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/deck/polybar-git/src/polybar/build/CMakeFiles/CMakeOutput.log".
See also "/home/deck/polybar-git/src/polybar/build/CMakeFiles/CMakeError.log".
==> ERROR: A failure occurred in build().
    Aborting...
```

Any ideas which package I need to install for this? 
`pacman -S xcb-util-image` did not help

I tried looking at https://github.com/polybar/polybar/wiki/Compiling#build-dependencies and installing all dependencies mentioned on there but I'm having trouble finding the right package for each of these in pacman / AUR. Is there perhaps an overview of all dependencies needed for Arch somewhere?

**Comment by patrick96** [comment, marked as answer]
The required dependencies are listed in the [AUR](https://aur.archlinux.org/packages/polybar-git). But the problem isn't that you haven't installed those dependencies, `makepkg -si` will install missing dependencies. It looks like the packages are already installed, but some of their files are missing, you likely have a bunch of broken packages in your system.

You can run `sudo pacman -Qqk` to list all packages that have missing files as well as which files are missing.

**Reply by MvEerd** [reply]
Ah ok, thanks for the clarification.

🤔  Seems like Valve is a bit eager to remove files they don't need from SteamOS then
```
❯ sudo pacman -Qqk | wc -l
88531
```
No idea where to begin even fixing all of these

`sudo pacman -Qknq | cut -d' ' -f 1 | sudo pacman -S -` may do the trick according to some searches. Will report back if it did

**Reply by MvEerd** [reply]
Reinstalling all packges with missing files using 
`sudo pacman -Qknq | cut -d' ' -f 1 | uniq | sudo pacman -S -` has solved this for me, I am now able to build polybar-git with `makepkg -si`

Thanks for your patience and time in debugging this, I hope this will be useful to anyone coming along trying to build on SteamOS
